app/controllers/taxon_framework_relationships_controller.rb

```diff
@@ -0,0 +1,203 @@
+    @relationships.delete_if{ |r| r.blank? }
+    @relationships = @relationships.map{ |r| r.gsub( "_"," " ) }
+    @taxon_framework = TaxonFramework.find_by_id( filter_params[:taxon_framework_id] ) unless filter_params[:taxon_framework_id].blank?    
+    @taxon_frameworks = TaxonFramework.all.limit( 100 )
+    @taxon = Taxon.find_by_id( filter_params[:taxon_id].to_i ) unless filter_params[:taxon_id].blank?
+    user_id = filter_params[:user_id] || params[:user_id]
+    @user = User.find_by_id(user_id) || User.find_by_login(user_id) unless user_id.blank?
+    @is_active = filter_params[:is_active]
+    @rank = filter_params[:rank] unless filter_params[:rank].blank?
+    
+    scope = TaxonFrameworkRelationship.all
+    scope = scope.relationships(@relationships) unless @relationships.blank?
+    scope = scope.taxon_framework(@taxon_framework) if @taxon_framework
+    scope = scope.taxon(@taxon) if @taxon
+    scope = scope.by(@user) if @user
+    scope = scope.rank(@rank) if @rank
+    if @is_active == "True"
+      scope = scope.active
+    elsif @is_active == "False"
+      scope = scope.inactive
+    end
+    
+    @taxon_framework_relationships = scope.page(params[:page]).
+      select("DISTINCT ON (taxon_framework_relationships.id) taxon_framework_relationships.*").
+      includes(:external_taxa ,taxa: [:taxon_names, :photos], taxon_framework: :taxon).
+      order("taxon_framework_relationships.id DESC")
+
+    respond_to do |format|
+      format.html
+      format.json do
+        taxon_options = { only: [:id, :name, :rank] }
+        external_taxon_options = { only: [:name, :url] }
+        render json: @taxon_framework_relationships.as_json(
+          methods: [:relationship],
+          include: [
+            { taxa: taxon_options },
+            { external_taxa: external_taxon_options }
+          ]
+        )
+      end
+    end
+  end
+  
+  def show
+    @taxon_framework = @taxon_framework_relationship.taxon_framework
+  end
+  
+  def new
+    if (taxon_framework_id = params['taxon_framework_id']).present?
+      @taxon_framework_relationship = TaxonFrameworkRelationship.new(taxon_framework_id: taxon_framework_id)
+    else
+      @taxon_framework_relationship = TaxonFrameworkRelationship.new
+    end
+    @taxon_framework_relationship.external_taxa.new
+    if ( taxon_id = params['taxon_id'] ).present? && taxon = Taxon.where( id: taxon_id ).first
+      @taxon_framework_relationship.taxa.new( id: taxon_id, rank: nil )
+      @taxon_frameworks = [TaxonFramework.joins( "JOIN taxa ON taxa.id = taxon_frameworks.taxon_id" ).where( "taxon_frameworks.rank_level <= ? AND taxon_id IN (?)", taxon.rank_level, taxon.ancestor_ids ).order( "taxa.rank_level ASC" ).first]
+    else
+      @taxon_framework_relationship.taxa.new
+      @taxon_frameworks = TaxonFramework.all.limit( 100 )
+    end
+  end
+  
+  def create
+    local_params = taxon_framework_relationship_params
+    taxa_attributes = local_params["taxa_attributes"]
```

Comment:
Is there a reason you're using strings as keys instead of symbols like we usually do?